ambry-frontend/src/main/java/com/github/ambry/frontend/PutBlobHandler.java

```diff
@@ -0,0 +1,405 @@
+     * @throws RestServiceException if there is an error while parsing the {@link BlobInfo} arguments.
+     */
+    private BlobInfo getBlobInfoFromRequest() throws RestServiceException {
+      long propsBuildStartTime = System.currentTimeMillis();
+      accountAndContainerInjector.injectAccountAndContainerForPutRequest(restRequest,
+          frontendMetrics.putBlobMetricsGroup);
+      BlobProperties blobProperties = RestUtils.buildBlobProperties(restRequest.getArgs());
+      Container container = RestUtils.getContainerFromArgs(restRequest.getArgs());
+      if (blobProperties.getTimeToLiveInSeconds() + TimeUnit.MILLISECONDS.toSeconds(
+          blobProperties.getCreationTimeInMs()) > Integer.MAX_VALUE) {
+        LOGGER.debug("TTL set to very large value in POST request with BlobProperties {}", blobProperties);
+        frontendMetrics.ttlTooLargeError.inc();
+      } else if (container.isTtlRequired() && (blobProperties.getTimeToLiveInSeconds() == Utils.Infinite_Time
+          || blobProperties.getTimeToLiveInSeconds() > frontendConfig.maxAcceptableTtlSecsIfTtlRequired)) {
+        String descriptor = RestUtils.getAccountFromArgs(restRequest.getArgs()).getName() + ":" + container.getName();
+        if (frontendConfig.failIfTtlRequiredButNotProvided) {
+          throw new RestServiceException(
+              "TTL < " + frontendConfig.maxAcceptableTtlSecsIfTtlRequired + " is required for upload to " + descriptor,
+              RestServiceErrorCode.InvalidArgs);
+        } else {
+          LOGGER.debug("{} attempted an upload with ttl {} to {}", blobProperties.getServiceId(),
+              blobProperties.getTimeToLiveInSeconds(), descriptor);
+          frontendMetrics.ttlNotCompliantError.inc();
+          restResponseChannel.setHeader(RestUtils.Headers.NON_COMPLIANCE_WARNING,
+              "TTL < " + frontendConfig.maxAcceptableTtlSecsIfTtlRequired + " will be required for future uploads");
+        }
+      }
+      // inject encryption frontendMetrics if applicable
+      if (blobProperties.isEncrypted()) {
+        restRequest.getMetricsTracker()
+            .injectMetrics(frontendMetrics.postBlobMetricsGroup.getRestRequestMetrics(restRequest.isSslUsed(), true));
+      }
+      byte[] userMetadata = RestUtils.buildUserMetadata(restRequest.getArgs());
+      frontendMetrics.blobPropsBuildTimeInMs.update(System.currentTimeMillis() - propsBuildStartTime);
+      LOGGER.trace("Blob properties of blob being POSTed - {}", blobProperties);
+      return new BlobInfo(blobProperties, userMetadata);
+    }
+
+    /**
+     * Enforce any additional requirements for certain types of uploads like data chunk uploads.
+     * @param blobProperties the {@link BlobProperties} parsed from the request.
+     * @throws RestServiceException
+     */
+    private void checkUploadRequirements(BlobProperties blobProperties) throws RestServiceException {
+      if (RestUtils.isChunkUpload(restRequest.getArgs())) {
```

Comment:
we can keep this for now, but I was thinking that we may not have to support chunk upload for put requests. I think those data chunks can always be uploaded as regular blobs (POST -> blob id) and then just the final stitchBlob call can be made with the put named blob API.

Reply:
Sorry I forgot to remove this part. Removed it.